library/Ivoz/Provider/Application/Service/ActiveCall/RegistrationChannelResolver.php

```diff
@@ -39,12 +39,15 @@ public function execute(?array $filters, array $tokenPayload): string
             $cr = $filters['cr'] ?? null;
             $dp = $filters['dp'] ?? null;
 
-            if ($direction === INBOUND && is_null($cr)) {
-                $cr = '*';
-            }
-
-            if ($direction === OUTBOUND && is_null($cr)) {
-                $dp = '*';
+            $canFilterByDirection = !is_null($direction) && is_null($cr) && is_null($dp);
+            if ($canFilterByDirection) {
+                if ($direction === 'inbound') {
+                    $cr = 'cr*';
+                }
+
+                if ($direction === 'outbound') {
+                    $dp = 'dp*';
```

Comment:
Actually the other way around. DDI Providers are used for inbound calls, Carrier for outbound calls. If you filter for outbound call you want to match all carriers (`cr*`);